### DISCUSSION TITLE
Error: Hydration error using Dark Mode with Next.js 15.0.2

### DISCUSSION THREAD

**Discussion by Rafacv23** [shadcn-ui/ui#5808] [Q&A] [answered]
This is my theme provider

```javascript
"use client"

import * as React from "react"
import { ThemeProvider as NextThemesProvider } from "next-themes"

export function ThemeProvider({
  children,
  ...props
}: React.ComponentProps<typeof NextThemesProvider>) {
  return <NextThemesProvider {...props}>{children}</NextThemesProvider>
}
```
My layout

```javascript
import type { Metadata } from "next"
import "./globals.css"
import Header from "@/components/Header"
import { ThemeProvider } from "@/components/ThemeProvider"
import Footer from "@/components/Footer"

export const metadata: Metadata = {
  title: "Bookmendator",
  description: "Generated by create next app",
}

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode
}>) {
  return (
    <html lang="en">
      <body>
        <ThemeProvider
          attribute="class"
          defaultTheme="system"
          enableSystem
          disableTransitionOnChange
        >
          <Header />
          {children}
          <Footer />
        </ThemeProvider>
      </body>
    </html>
  )
}
```

I have this error:

![image](https://github.com/user-attachments/assets/6f164b82-5ff9-43b7-a029-c8274e5b3597)

**Comment by Smnthjm08** [comment, marked as answer]
add this for the html tag in the root layout ` <html lang="en" suppressHydrationWarning>` . This should fix it.

**Reply by Rafacv23** [reply]
It works! thanks

**Reply by agamm** [reply]
Is this really the right solution to just ignore it?

**Comment by artysan-code** [comment]
I don't think the solution it's to suppress the Warning

**Comment by Shirshakkandel** [comment]
As we know code first run on server side which don't have our system preference. Nextjs only know our system preference when code is run in client. So we need to make dark mode and light mode code client side render for this you can create new component ClientRender and wrap <ThemeProvider> 

for video reference => [youtube video reference](https://www.youtube.com/watch?v=RFf1dkzm-T0&ab_channel=shirshakKandel
)
```typescript
//src\components\ClientRender.tsx
'use client';

import React, { ReactNode, useEffect, useState } from 'react';

export const RenderMounted = ({ children }: { children: ReactNode }) => {
  const [mounted, setMounted] = useState(false);

  useEffect(() => setMounted(true), []);

  if (!mounted) return null;

  return <>{children}</>;
};

```


```typescript
  <ClientRender>
          <ThemeProvider
            attribute='class'
            defaultTheme='light'
            // enableSystem
            // disableTransitionOnChange
          >
            <ObserverProvider>{children}</ObserverProvider>
          </ThemeProvider>
        </ClientRender>
```

After this code will only run on client when client have proper knowledge of system preference of user removing hydration error.

**Reply by cygullem** [reply]
Hey! 👋 I noticed something in your example that might be worth clarifying. In the file named `ClientRender.tsx`, you defined and exported a component called `RenderMounted`, but in your layout or usage example, you used `<ClientRender>` instead of `<RenderMounted>`.

**Reply by dhirajaryaa** [reply]
@Shirshakkandel your solution is work but problem is hole app code server send in script not in html tag [one line : app become client side]
